cmd/compgen/READEME.md

```diff
@@ -0,0 +1,17 @@
+# Compgen
+
+Compgen is a tool for building new otel components.
```

Comment:
This could be more verbose and specifically calling out what components it's able to build at this time.